packages/xo-server/src/xapi-stats.mjs

```diff
@@ -226,31 +221,25 @@ const STATS = {
 //   data: Item[columns] // Item = { t: Number, values: Number[rows] }
 // }
 
-// Local cache
-// _statsByObject : {
-//   [uuid]: {
-//     [step]: {
-//       endTimestamp: Number, // the timestamp of the last statistic point
-//       interval: Number, // step
-//       stats: {
-//         [metric1]: Number[],
-//         [metric2]: {
-//           [subMetric]: Number[],
-//         }
-//       }
-//     }
-//   }
-// }
 export default class XapiStats {
+  // hostCache => host uid => granularity =>  {
+  //  timestamp
+  //  value : promise or value
+  // }
+  #hostCache = {}
   constructor() {
     this._statsByObject = {}
   }
 
   // Execute one http request on a XenServer for get stats
   // Return stats (Json format) or throws got exception
-  @limitConcurrency(3)
   _getJson(xapi, host, timestamp, step) {
-    return xapi
+    const byHost = this.#hostCache[host.uuid]?.[step]
+    if (byHost !== undefined && byHost.timestamp + step > timestamp) {
+      return byHost.value
+    }
+    this.#hostCache[host.uuid] = this.#hostCache[host.uuid] ?? {}
```

Comment:
Not a big deal, but ideally we should clear the staled stats from memory to avoid keeping entries for no longer existing hosts forever.